src/components/SearchForm.vue

```diff
@@ -0,0 +1,38 @@
+<template>
+  <form @submit.prevent>
+    <input
+      type="text"
+      name="breed"
+      id="breed"
+      placeholder="Type a dog breed..."
+      ref="searchInput"
+      @input="debounceSearch"
+    />
+  </form>
+</template>
+
+<script>
+export default {
+  data() {
+    return {
+      searchInput: "",
+      debounce: null,
+    };
+  },
+  methods: {
+    debounceSearch(event) {
+      clearTimeout(this.debounce);
+      this.debounce = setTimeout(() => {
+        this.searchInput = event.target.value;
+        this.$emit("search", this.searchInput);
+      }, 1000);
+    },
```

Comment:
dodal sem debounce fn s katero uporabnik ko išče po pasmi, ne fetcha podatko za vsak znak ampak, ko user vpiše zadnji znak in ko mine 1s po zadnjem vpisanem znaku, le ta vrednost se emita potem v breedsList komponento, kjer se uporabi v getSearchResult kot input za fetchanje.

Reply:
super 👍 

kasneje lahko narediš `debounce` kot svojo utils funkcijo da je re-usable.. tko da notr pasaš funkcijo k jo hočeš "debouncat"..

Reply:
@Luka85 pol pa če vse dela tko k more, lahko zmergaš tole 👍

Reply:
> super 👍
> 
> kasneje lahko narediš `debounce` kot svojo utils funkcijo da je re-usable.. tko da notr pasaš funkcijo k jo hočeš "debouncat".. in pa še da to sekundo zmanjšaš, ker pomoje je to mal preveč, da user čaka eno sekundo da se api request sploh nardi.. ponavad se uporablja nekih 200-300 milisekund za take stvari 👍

Ok, popravim :) nism mel občutka kok je ponavadi dejansko ;) 
Sej ta mi še ni čist jasna, če sm iskren, bom probov, še mal preštudirat oz še na kakšnem drugem primeru nardit.
Hotel sem še narediti, da ko se stran naloži, in če se podatki ne naložijo,da se  input disable-a, ker sedaj lahko še vseeno išče podatke, tudi če vseh breedsov ni naloženih